connector-sdk/core/src/main/java/io/camunda/connector/api/inbound/CorrelationResult.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Copyright Camunda Services GmbH and/or licensed to Camunda Services GmbH
+ * under one or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information regarding copyright
+ * ownership. Camunda licenses this file to you under the Apache License,
+ * Version 2.0; you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.camunda.connector.api.inbound;
+
+public sealed interface CorrelationResult {
+
+  final class Success implements CorrelationResult {
+
+    public static final Success INSTANCE = new Success();
+  }
+
+  record Failure(ErrorCode code, String message, Throwable error) implements CorrelationResult {}
+
+  enum ErrorCode {
+    INVALID_INPUT(false),
+    ACTIVATION_CONDITION_NOT_MET(false),
+    FAULT_ZEEBE_CLIENT_STATUS(true),
+    FAULT_IDEMPOTENCY_KEY(false),
```

Comment:
Should `FAULT_IDEMPOTENCY_KEY` indicate an already existing message?

Reply:
No, I think it's for when we can't extract the idempotency key from the payload (i.e. no message id / message id expression)

Reply:
Shouldnt this rather be an INVALID_INPUT error then?

Reply:
And another one: We return a success in case of an already existing message right?

Reply:
No, as far as I know this would raise a ClientStatusException as the message would be rejected ([src](https://docs.camunda.io/docs/components/concepts/messages/#message-uniqueness)).

I'm ok with removing this error code - will do. I just copied it as it was initially.

Reply:
Yes, we have to. What about a dedicated class `Correlation.Failure.Zeebe` that allows to inspect the actual zeebe correlation error?

Reply:
We might also want to be able to provide access to the exception for `Unknown` WDYT?

Reply:
We could offer `Correlation.Failure.Exception(e)`

Reply:
I would name it `Correlation.Failure.Other` - Exception is probably not the best name as it conflicts with the standard Java Exception class.

Reply:
Reworked the sealed hierarchy as discussed